python/hsfs/core/tags_api.py

```diff
@@ -54,8 +54,7 @@ def add(self, metadata_instance, name, value):
             "tags",
             name,
         ]
-        query_params = {"value": value} if value else None
-        _client._send_request("PUT", path_params, query_params=query_params)
+        _client._send_request("PUT", path_params, data=value)
```

Comment:
Did you test this? You are not setting the JSON header. What happens if the value provided by the user is not a valid JSON?

Reply:
Added the header. 
The backend deals with malformed tags - malformed json or value that does not match schema.